sdk/servicebus/azure-servicebus/tests/perf_tests/T1_legacy_tests/_test_base.py

```diff
@@ -0,0 +1,148 @@
+    async_queue_client = None
+
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        connection_string = self.get_from_env("AZURE_SERVICEBUS_CONNECTION_STRING")
+        connection_props = parse_connection_string(connection_string)
+        self.mgmt_client = ServiceBusService(
+            service_namespace=connection_props['namespace'],
+            shared_access_key_name=connection_props['shared_access_key_name'],
+            shared_access_key_value=connection_props['shared_access_key'])
+
+    async def global_setup(self):
+        await super().global_setup()
+        queue = Queue(max_size_in_megabytes=40960)
+        self.mgmt_client.create_queue(self.queue_name, queue=queue)
+
+    async def setup(self):
+        await super().setup()
+        # In T1, these operations check for the existance of the queue
+        # so must be created during setup, rather than in the constructor.
+        self.queue_client = self.service_client.get_queue(self.queue_name)
+        self.async_queue_client = self.async_service_client.get_queue(self.queue_name)
+
+    async def global_cleanup(self):
+        self.mgmt_client.delete_queue(self.queue_name)
+        await super().global_cleanup()
+
+
+class _SendTest(_QueueTest):
+    sender = None
+    async_sender = None
+
+    async def setup(self):
+        await super().setup()
+        self.sender = self.queue_client.get_sender()
+        self.async_sender = self.async_queue_client.get_sender()
+        self.sender.open()
+        await self.async_sender.open()
+    
+    async def close(self):
+        self.sender.close()
+        await self.async_sender.close()
+        await super().close()
+
+
+class _ReceiveTest(_QueueTest):
+    receiver = None
+    async_receiver = None
+
+    async def global_setup(self):
+        await super().global_setup()
+        await self._preload_queue()
+
+    async def setup(self):
+        await super().setup()
+        mode = ReceiveSettleMode.PeekLock if self.args.peeklock else ReceiveSettleMode.ReceiveAndDelete
+        self.receiver = self.queue_client.get_receiver(
+            mode=mode,
+            prefetch=self.args.num_messages,
+            idle_timeout=self.args.max_wait_time)
+        self.async_receiver = self.async_queue_client.get_receiver(
+            mode=mode,
+            prefetch=self.args.num_messages,
+            idle_timeout=self.args.max_wait_time)
+        self.receiver.open()
+        await self.async_receiver.open()
+
+    async def _preload_queue(self):
+        data = get_random_bytes(self.args.message_size)
+        async_queue_client = self.async_service_client.get_queue(self.queue_name)
+        async with async_queue_client.get_sender() as sender:
+            for i in range(self.args.preload):
+                sender.queue_message(Message(data))
+                if i % 1000 == 0:
+                    print("Loaded {} messages".format(i))
+                    await sender.send_pending_messages()
```

Comment:
if message_size * 1000 exceeds max allowed size on the link, then we'll get an error.

Reply:
I can't remember whether the T1 SDK handled this already - but given that this is just for adding messages to the queue before running the test, I'm not too concerned if an error is raised here. I've tried running this test with large messages and I don't think It's caused issues yet.

Reply:
aha yes, I was thinking wrong, I thought they would go in a single batch message but they are separate messages. so it should be good.
sorry about making the noise :P